skiko/src/awtMain/kotlin/org/jetbrains/skiko/swing/SwingOffscreenDrawer.kt

```diff
@@ -56,7 +58,13 @@ internal class SwingOffscreenDrawer(
         dirtyRectangles: List<Rectangle>
     ): BufferedImage {
         val src = ByteBuffer.wrap(bytes)
-        val image = BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB_PRE)
+        if (bufferedImage == null || bufferedImage?.width != width || bufferedImage?.height != height) {
+            bufferedImage = BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB_PRE)
```

Comment:
Possible additional optimizations:
1. Call `flush()` on the old `bufferedImage` to clear the GPU memory without waiting for GC.
2. (Questionable) If the new size is smaller on both dimensions, you could reuse the `BufferedImage` by using `getSubimage()`.

Reply:
> I'm not sure if optimization 2 is worth the extra code complexity, since a one-off allocation is (relatively) cheap.

You could have many allocations in a short period when the window is being resized. But using `getSubImage()` by itself won't solve it if the window is getting bigger. What you could do, though, is to allocate a slightly bigger image upfront (say 20% larger on each axis) and use `getSubImage()` on that.

Reply:
Sounds like a nice follow-up, but we can probably do with the fix as-is for now? Should cover most cases already causing issues.